packages/api-client/src/main/user/UserAPI.ts

```diff
@@ -409,7 +409,7 @@ export class UserAPI {
    * @param propertyData The property data to set
    * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/setProperty
    */
-  public async putProperty(propertyKey: string, propertyData: Object): Promise<void> {
+  public async putProperty<T extends string>(propertyKey: T, propertyData: Record<T, any>): Promise<void> {
```

Comment:
This means that if you call `getProperty('foo')` the resulting object will be `{ foo: ... }`.